tensorboard/plugins/hparams/api.proto

```diff
@@ -0,0 +1,187 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+// Defines a proto3-based REST API that the HParams web-component of the plugin
+// would use to read data from a hyperparameter-tuning experiment.
+// The first part of this file defines the message types (resources) used
+// to pass information into and out of the API methods. These messages will be
+// transmited using their JSON encoding. The second part documents the actual
+// API and HTTP end-points.
+
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+
+package tensorboard.plugins.hparams;
+
+// General note: in what follows we use the the field 'name' of a message to
+// stores its id. We avoid calling this field 'id' since it is a reserved word
+// in python and to be compliant with the API style guide detailed in
+// https://cloud.google.com/apis/design/.
+
+
+// Represents a single experiment.
+// An experiment consists of multiple "sessions". Typically, in each session
+// a model is trained for a given set of hyperparameter values. In each session
+// a training program may generate one or more series of real
+// numbers--each containing the evaluation of some metric at different
+// training steps.
+//
+// In some experiments, the user trains a model with the same set of
+// hyperparameters multiple times to get the distribution of metric
+// evaluations when the computation (such as the training algorithm, or metric
+// evaluation) is non-deterministic. To make the UI aware of this, sessions
+// are partitioned into groups: each group consists of all training sessions
+// which share the same values for the hyperparameters. In experiments with no
+// repeated executions, each group consists of exactly one session.
+message Experiment {
```

Comment:
I assume that experiments are uniquely identified by a tuple of (name, user), or does that not have to be the case?
Will the UI display experiments for one user at a time, or do we envision displaying experiments across users (in that case, experiment names could clash)?

Reply:
Well, currently, as Nick mentions below, there is only one experiment in each open Tensorboard server, so there really is no need to have an experiment id. Let me remove and add it later when we think about multiple experiments.